addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -310,6 +342,249 @@ def test_discount_max_amount_on_specific_product(self):
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.product_A.id,
+                    'tax_id': False,
+                }),
+            ]
+        })
+
+        promotion_program = self.env['loyalty.program'].create([{
+            'name': 'Coupon Program',
+            'program_type': 'promotion',
+            'trigger': 'auto',
+            'applies_on': 'current',
+            'rule_ids': [Command.create({
+                    'reward_point_amount': 1,
+                    'reward_point_mode': 'order',
+                    'minimum_amount': 10,
+                })],
+            'reward_ids': [Command.create({
+                    'reward_type': 'discount',
+                    'discount': 10.0,
+                    'discount_applicability': 'order',
+                    'required_points': 1,
+                })],
+        }])
+
+        self.assertEqual(order.amount_total, 100)
+        self._auto_rewards(order, promotion_program)
+        self.assertEqual(order.amount_total, 90)
+        order.action_confirm()
+        self.assertEqual(loyalty_card.points, 90)
+
+    def test_points_awarded_discount_code_no_domain_program(self):
+        """
+        Check the calculation for points awarded when there is a discount coupon applied and the
+        loyalty program applies on all product (no domain).
+        """
+        loyalty_program = self.create_programs([('Loyalty P', 'loyalty')])['Loyalty P']
+        loyalty_card = self.env['loyalty.card'].create({
+            'program_id': loyalty_program.id,
+            'partner_id': self.partner_a.id,
+            'points': 0,
+        })
+
+        order = self.env['sale.order'].with_user(self.user_salemanager).create({
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.product_A.id,
+                    'tax_id': False,
+                }),
+            ]
+        })
+
+        self.assertEqual(order.amount_total, 100)
+        self._apply_promo_code(order, "test_10pc")
+        self.assertEqual(order.amount_total, 90)
+        order.action_confirm()
+        self.assertEqual(loyalty_card.points, 90)
+
+    def test_points_awarded_general_discount_code_specific_domain_program(self):
+        """
+        Check the calculation for points awarded when there is a discount coupon applied and the
+        loyalty program applies on a sepcific domain. The discount code has no domain. The product
+        related to that discount is not in the domain of the loyalty program.
+        Expected behavior: The discount is not included in the computation of points
+        """
+        product_category_base = self.env.ref('product.product_category_1')
+        product_category_food = self.env['product.category'].create({
+            'name': 'Food',
```

Comment:
Nit (same for the next test):
```suggestion
            'name': "Food",
```